sympy/solvers/solvers.py

```diff
@@ -513,31 +529,53 @@ def solve(f, *symbols, **flags):
 
         * To solve for a function within a derivative, use ``dsolve``.
 
-    Single expression and more than one symbol:
+    Single expression and more than one symbol.
 
-        * When there is a linear solution:
+        Some care must be taken here since, currently, SymPy might try
+        to treat this as an attempt to match coefficients instead of
+        as a request to give solutions for as many of the symbols as
+        possible.
 
-            >>> solve(x - y**2, x, y)
-            [(y**2, y)]
-            >>> solve(x**2 - y, x, y)
-            [(x, x**2)]
-            >>> solve(x**2 - y, x, y, dict=True)
-            [{y: x**2}]
+        You will know when the case of undetermined coefficients has been
+        identified by Sympy because a singe dictionary will be returned.
+        Here, it might be obvious that one is trying to match coefficients.
+
+            >>> eq_ab = Eq(a*x + b, 2*x - 3)
+            >>> solve(eq_ab, a, b)
+            {a: 2, b: -3}
 
-        * When undetermined coefficients are identified:
+        But that might not be what you wanted. To disable that behavior,
+        set the flag ``undetermined=False``. In the future, the only
+        way to get this behavior will be to call directly the currently
+        available function, `solve_undetermined_coeffs`.
```

Comment:
This could be ma de a sphinx reference so that it generates a clickable link in the html docs.

Reply:
I guess also in the future the `undetermined` flag will become a no-op.